astropy/config/paths.py

```diff
@@ -79,7 +53,7 @@ def get_config_dir_path(rootname: str = "astropy") -> Path:
         The absolute path to the configuration directory.
 
     """
-    return _get_dir_path(rootname, set_temp_config, "config")
+    return set_temp_config.get_dir_path(rootname)
```

Comment:
I am not sure about using `set_temp_config` here. It was meant for temporary stuff, not suitable for cases like you want to permanently change the path for a machine.

Currently your patch only deals with cache, but config also should be consistent:

```
HOME
  |__ .astropy
        |___ cache
        |___ config
```

I think the ask was to change where HOME points to, which affects both cache and config.

What I had in mind, I think, was simpler, we add one more `if` in ` _get_dir_path` to check for the new env var first before going to XDG env var (for both cache and config).

Additionally, the user doc would need to make clear of this new option and how it might catch users unawares if they accidentally have those new env var defined, etc.

Also have to make sure we don't break https://docs.astropy.org/en/latest/config/index.html#customizing-config-location-in-affiliated-packages in the process.

Reply:
This bit is from #17554. I'll move your comment and answer there !